app/component/ItineraryTab.js

```diff
@@ -126,13 +126,23 @@ class ItineraryTab extends React.Component {
 
   render() {
     const { itinerary } = this.props;
+    let { fares } = itinerary;
     const { config } = this.context;
 
     if (!itinerary || !itinerary.legs[0]) {
       return null;
     }
 
-    const fares = getFares(itinerary.fares, getRoutes(itinerary.legs), config);
+    if (fares === null && config.URL.FARES) {
+      fetchFares(itinerary, config.URL.FARES)
```

Comment:
@patrickbeck1337 @andreashelms Could you review this part give advice on what the correct React way of fetching the price information is? Right now the information is fetched whenever the component is rendered.

Reply:
I made some refactoring - moved the fetchFares function to the componentDidMount(). This made one test break, which I'm aware of, I only pushed for you all to see the current state.

Reply:
You could check if `fares` received via props differ from `fares` in state and only call fetchFares if thats the case.

Reply:
And you should do that in the ordinary `render` method?

Reply:
Or right here in `componentDidMount` where `fetchFares` is executed. You could also do that within `UNSAFE_componentWillReceiveProps(nextProps)` lifecycle function and compare nextProps with old ones in order to decide if fares need to be fetched. Even better would be to use lifecycle hooks (e.g. useEffect) but that would mean to refactor the whole component to a functional component.

Reply:
If the `fetchFares()` function is inside the `componentDidMount` it will only run once, and based on this I don't believe that comparing `props.itinerary.fares` with `state.fares` would be necessary since after a refresh, the state would be also reinitialized. Am I right? I'd keep the `props.itinerary.fares` null check and only fetch if the component did not get any fare info. Let me know if I'm on a wrong track.

Reply:
You are totally right. If It only needs to be run once, it's placed correctly here. I checked out your branch and it works well. I don't see any need for refactoring here.